src/screens/Collectible/Collectible.js

```diff
@@ -97,20 +103,58 @@ const CollectibleImage = styled(CachedImage)`
 const iconSend = require('assets/icons/icon_send.png');
 const genericCollectible = require('assets/images/no_logo.png');
 
-class CollectibleScreen extends React.Component<Props> {
-  shouldComponentUpdate(nextProps: Props) {
+class CollectibleScreen extends React.Component<Props, State> {
+  state = {
+    isImageViewVisible: false,
+  };
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
     const isFocused = this.props.navigation.isFocused();
     if (!isFocused) {
       return false;
     }
-    const isEq = isEqual(this.props, nextProps);
-    return !isEq;
+    if (!isEqual(this.state, nextState)) {
+      return true;
+    }
+    if (!isEqual(this.props, nextProps)) {
+      return true;
+    }
+    return false;
   }
 
   goToSendTokenFlow = (assetData: Object) => {
     this.props.navigation.navigate(SEND_COLLECTIBLE_FROM_ASSET_FLOW, { assetData });
   };
 
+  onTouchImage = () => {
+    this.setState({ isImageViewVisible: true });
+  }
+
+  onCloseImageView = () => {
+    this.setState({ isImageViewVisible: false });
+  }
+
+  renderImageView(collectible: Collectible) {
+    const { isImageViewVisible } = this.state;
+    const { image, name } = collectible;
+
+    const imageViewImages = [
+      {
+        source: { uri: image },
+        title: name,
+      },
+    ];
+
+    return (
+      <ImageView
+        images={imageViewImages}
+        imageIndex={0}
+        isVisible={isImageViewVisible}
+        onClose={this.onCloseImageView}
```

Comment:
Would be great to add close icon that we use on other modals. There's a way to add custom close component using controls prop: https://www.npmjs.com/package/react-native-image-view#controls-prop

Reply:
Updated